src/hooks/useChatSocket.tsx

```diff
@@ -39,14 +52,29 @@ export const useChatSocket = () => {
             console.log("Waiting to connect");
         })
 
-        newSocket.on(ServerEvents.MATCHED, ({ partnerName }) => {
+        newSocket.on(ServerEvents.MATCHED, async ({ partnerName, partnerPk }) => {
             setPartner(partnerName);
+
+            const partnerPublicK = await importPublicKey(partnerPk);
+            await storeKey(KeyTransaction.PARTNER_PK, partnerPublicK);
+
             setIsWaiting(false);
             console.log("Matched with a Partner");
             setMessages([]);
         });
 
-        newSocket.on(ServerEvents.RECEIVE_MESSAGE, (data) => {
+        newSocket.on(ServerEvents.RECEIVE_MESSAGE, async (data) => {
+
+            const partnerPk = await getKey(KeyTransaction.PARTNER_PK);
+            const selfPk = await getKey(KeyTransaction.SELF_PK);
+            const sharedSecret = await deriveSharedSecret(selfPk!!, partnerPk!!);
```

Comment:
I don't think the double exclamation `!!` means anything here. If you are using a non-null assertion, it should be just `!`